### PR TITLE
Refactor(Processing): Improve job mapping API

### PR DESCRIPTION
**Pull request by gkillough**

### CODE REVIEW

File: processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/model/FilteredJobNotificationWrapper.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.UUID;
+
+import com.synopsys.integration.alert.common.enumeration.ProcessingType;
+import com.synopsys.integration.alert.common.rest.model.AlertSerializableModel;
+
+public class FilteredJobNotificationWrapper extends AlertSerializableModel {
+    private final UUID jobId;
+    private final ProcessingType processingType;
+    private final String channelName;
+    private final List<FilterableNotificationWrapper> jobNotifications;
```
Comment (bamandel) [inline processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/model/FilteredJobNotificationWrapper.java:35]: Do we need the entire FilterableNotificationWrapper? Can we get away with a new model that doesn't have all the filtering data?

File: processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/JobNotificationMapper.java [hunk 2]
```diff
@@ -73,7 +73,15 @@ public DefaultJobNotificationExtractor(JobAccessor jobAccessor) {
...
             }
         }
-        return groupedFilterableNotifications;
+
+        List<FilteredJobNotificationWrapper> filterableJobNotifications = new LinkedList<>();
+        for (Map.Entry<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper>> groupedEntry : groupedFilterableNotifications.entrySet()) {
+            FilteredDistributionJobResponseModel filteredJob = groupedEntry.getKey();
+            FilteredJobNotificationWrapper wrappedJobNotifications = new FilteredJobNotificationWrapper(filteredJob.getId(), filteredJob.getProcessingType(), filteredJob.getChannelName(), groupedEntry.getValue());
+            filterableJobNotifications.add(wrappedJobNotifications);
+        }
```
Comment (bamandel) [inline processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/JobNotificationMapper.java:82]: are we sure we want this additional processing into a list? It looks like an extra iteration for not too much of a benefit

Comment (gkillough) [reply processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/JobNotificationMapper.java:82]: The benefit is a clearer API. We could optimize down to one iteration, but the concern this is addressing is to no longer return a `Map`.

File: processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/model/FilteredJobNotificationWrapper.java (hunk 1 above)
Comment (gkillough) [reply processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/model/FilteredJobNotificationWrapper.java:35]: Maybe not. This was just flattening the `Map.Entry`. I'll investigate.

File: processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationProcessorV2.java [hunk 3]
```diff
@@ -90,23 +90,23 @@ public final void processNotifications(List<AlertNotificationModel> notification
...
                                                                           .flatMap(Optional::stream)
                                                                           .collect(Collectors.toList());
-        Map<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper>> jobsToNotifications = jobNotificationExtractor.mapJobsToNotifications(filterableNotifications, frequencies);
-        for (Map.Entry<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper>> jobToNotificationPair : jobsToNotifications.entrySet()) {
-            FilteredDistributionJobResponseModel filteredDistributionJobResponseModel = jobToNotificationPair.getKey();
-            List<FilterableNotificationWrapper> filteredNotifications = jobToNotificationPair.getValue();
+        List<FilteredJobNotificationWrapper> mappedNotifications = jobNotificationMapper.mapJobsToNotifications(filterableNotifications, frequencies);
+        for (FilteredJobNotificationWrapper jobNotificationWrapper : mappedNotifications) {
+            UUID jobId = jobNotificationWrapper.getJobId();
```
Comment (bamandel) [inline processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationProcessorV2.java:95]: Small thing but we could just call getJobId when we use it on line 102

**Review by bamandel — APPROVED** [review]

**Pull request merged**